### PR TITLE
Starter kit

### PR DESCRIPTION
**Pull request by dominic-blain**
There's a _lot_ going on here. Here's a non-exhaustive list of what has been changed:

- New form files
- Added basic modules
- Added basic buttons
- Updated site-header/footer
- Updated site-nav
- Added some recipes (will need some more work later)
- Added new way to manage intervals between modules (util-module-interval)
- Basic markdown styles + improved markdown rewriting
- Added basic site loading
- New fluid theme (will need some tuning and a way to quickly revert to regular theme)
- New breakpoints that contain popular screen size instead of being exactly on popular sizes
- Added Picture template
- Revamp of page title / master title by @hotdoy 
- Revamp of Metas by @hotdoy 
- Cleanup of icons
- Revamp of share button with native sharing
- Added basic social links
- Added new copyright template
- Renamed popup to modal

### CODE REVIEW

File: workspace/assets/js.json [hunk 1]
```diff
@@ -28,12 +35,21 @@
...
+		"modules/auto-video.js",
 		"modules/keyboard-nav.js",
 		"modules/tab-navigation.js",
 		"modules/auto-tracked-scroll.js",
 		"modules/auto-change-state-click.js",
+		"modules/auto-native-share.js",
+		"modules/auto-copy-on-click.js",
 		"modules/link-selector.js",
+		"modules/watermark.js",
+		"modules/form-dyn.js",
```
[the remark this answers is not in the document]
Comment (fhamon) [reply workspace/assets/js.json]: After talking with @nitriques no form code should be included by deafault in js.json.

File: workspace/assets/css/site/util-custom.less [hunk 2]
```diff
@@ -1,3 +1,37 @@
+//=========================
+// = MISSING CLASSES 
+//=========================
+
+.appearance-none {
```
[the remark this answers is not in the document]
Comment (dominic-blain) [reply workspace/assets/css/site/util-custom.less]: This will be done later (or not) by @PascalPiche

File: workspace/assets/js.json [hunk 3]
```diff
@@ -54,7 +62,10 @@
 		"https://cdnjs.cloudflare.com/ajax/libs/jquery-easing/1.3/jquery.easing.min.js",
 		"https://cdnjs.cloudflare.com/ajax/libs/underscore.js/1.8.3/underscore-min.js",
 		"https://cdnjs.cloudflare.com/ajax/libs/underscore.string/3.3.4/underscore.string.min.js",
-		"https://cdnjs.cloudflare.com/ajax/libs/velocity/1.5.0/velocity.min.js"
+		"https://cdnjs.cloudflare.com/ajax/libs/velocity/1.5.0/velocity.min.js",
+		"https://www.google.com/recaptcha/api.js?onload=GoogleReCaptchaCallback&amp;render=explicit",
```
Comment (fhamon) [inline workspace/assets/js.json]: They should not be there by default

File: workspace/assets/js.json [hunk 4]
```diff
@@ -5,11 +5,15 @@
...
-		
 		"fx/tracking.js",
 		
 		"com/article-changer-v3.js",
 		"com/checkpoint-event.js",
+		"com/oembed.js",
+		"com/video.js",
+
+		"modules/oembed-yt.js",
+		"modules/oembed-vm.js",
```
Comment (fhamon) [inline workspace/assets/js.json]: They should not be there by default

File: workspace/utilities/lib/util-ficelle.xsl [hunk 5]
```diff
@@ -0,0 +1,23 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
[the remark this answers is not in the document]
Comment (fhamon) [reply workspace/utilities/lib/util-ficelle.xsl]: We can leave it.

**Comment by nitriques** [comment]
@dominic-blain can't merge, please rebase...

**Comment by dominic-blain** [comment]
Done! @nitriques

File: install/includes/config_default.php [hunk 6]
```diff
@@ -0,0 +1,127 @@
+<?php
```
Comment (nitriques) [inline install/includes/config_default.php]: this file should be in manifest folder, and should be named unattended.php

You'll thank me later.

File: workspace/assets/css/vars/theme.xml [hunk 7]
```diff
@@ -5,23 +5,126 @@
 
 		<!-- Colors -->
 		<colors>
+			<!-- Brand -->
+			<type name="symbolic">
```
Comment (nitriques) [inline workspace/assets/css/vars/theme.xml]: Can we comment out most of them ? If not, it will create a big ass css file...

File: workspace/assets/js.json (hunk 4 above)
Comment (nitriques) [reply workspace/assets/js.json]: @dominic-blain yes please remove and keep the only basic basic minimum.

File: workspace/assets/js.json (hunk 3 above)
Comment (nitriques) [reply workspace/assets/js.json]: No please remove. It is also not at the right place.

File: workspace/assets/js/dev/panel.js [hunk 8]
```diff
@@ -103,7 +103,7 @@
 	};
 
 	var initDevPanel = function () {
-		var storage = global.AppStorage.session;
+		var storage = global.App.storage.session;
```
[the remark this answers is not in the document]
Comment (nitriques) [reply workspace/assets/js/dev/panel.js]: Yes.

File: workspace/assets/js/utils/global.js [hunk 9]
```diff
@@ -1,193 +0,0 @@
...
-		window.msCancelRequestAnimationFrame || window.clearTimeout;
-	
-	// default easing support
-	$.easing.def = ($.mobile ? 'linear' : 'easeOutQuad');
-})(jQuery, window);
-
-/**
- * Cheap modrnzr
- */
-(function ($, global, undefined) {
```
[the remark this answers is not in the document]
Comment (nitriques) [reply workspace/assets/js/utils/global.js]: yes please keep the file

File: workspace/assets/package.json [hunk 10]
```diff
@@ -1,6 +1,6 @@
 {
-  "name": "project-name",
-  "title": "ProjectTitle",
+  "name": "starterkit",
```
[the remark this answers is not in the document]
Comment (nitriques) [reply workspace/assets/package.json]: Yes, please revert

File: workspace/data-sources/data.navigation.php [hunk 11]
```diff
@@ -0,0 +1,42 @@
+<?php
+
+require_once(EXTENSIONS.'/page_lhandles/lib/class.datasource.MultilingualNavigation.php');
```
Comment (nitriques) [inline workspace/data-sources/data.navigation.php]: This will crash if the extension is not present. We should keep the old commented part.

File: workspace/data-sources/data.pages_403.php [hunk 12]
```diff
@@ -0,0 +1,62 @@
+<?php
+
```
Comment (nitriques) [inline workspace/data-sources/data.pages_403.php]: Pumping the 403 and 404 from the page section is not a good idea. Please remove this.

File: workspace/data-sources/data.pages_404.php [hunk 13]
```diff
@@ -0,0 +1,63 @@
+<?php
```
Comment (nitriques) [inline workspace/data-sources/data.pages_404.php]: Pumping the 403 and 404 from the page section is not a good idea. Please remove this.

File: workspace/data-sources/data.pages_detail.php [hunk 14]
```diff
@@ -0,0 +1,65 @@
...
+    public $dsParamREQUIREDPARAM = '$handle';
+    public $dsParamPARAMXML = 'no';
+    public $dsParamSORT = 'system:id';
+    public $dsParamHTMLENCODE = 'no';
+    public $dsParamASSOCIATEDENTRYCOUNTS = 'no';
+
+    public $dsParamFILTERS = array(
+        '26' => 'handle: {$handle}',
+        '36' => '{$login-filter}',
+        '183' => 'not: Index',
```
Comment (nitriques) [inline workspace/data-sources/data.pages_detail.php]: Boy do I hate this.

File: workspace/data-sources/data.pages_erreurs.php [hunk 15]
```diff
@@ -1,74 +0,0 @@
-<?php
```
Comment (nitriques) [inline workspace/data-sources/data.pages_erreurs.php]: Please keep this.

File: workspace/data-sources/data.pages_index.php [hunk 16]
```diff
@@ -0,0 +1,64 @@
+<?php
```
Comment (nitriques) [inline workspace/data-sources/data.pages_index.php]: Boy do I hate this.

File: workspace/jit-image-manipulation/trusted-sites [hunk 17]
```diff
@@ -1,34 +0,0 @@
-i.ytimg.com/*
```
Comment (nitriques) [inline workspace/jit-image-manipulation/trusted-sites]: You clearly are deleting something you do not understand ;)

Please keep this.

File: workspace/pages/403.xsl [hunk 18]
```diff
@@ -2,11 +2,15 @@
...
+	doctype-public="-//W3C//DTD XHTML 1.0 Strict//EN"
+	doctype-system="http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd"
+	omit-xml-declaration="yes"
+	encoding="UTF-8"
+	indent="yes" />
 
-<xsl:template match="data">
-	<xsl:apply-templates select="pages-erreurs/entry[1]" />
+<xsl:template match="/">
+	<h1><xsl:value-of select="/data/params/page-title"/></h1>
```
[the remark this answers is not in the document]
Comment (nitriques) [reply workspace/pages/403.xsl]: Does not seems like it, and its the worst idea of 2019. Please revert.

File: workspace/pages/404.xsl [hunk 19]
```diff
@@ -2,11 +2,15 @@
 <xsl:stylesheet version="1.0"
 	xmlns:xsl="http://www.w3.org/1999/XSL/Transform">
 
-<xsl:import href="../utilities/master/master.xsl" />
-<xsl:import href="../utilities/com/erreurs.xsl" />
+<xsl:output method="xml"
```
Comment (nitriques) [inline workspace/pages/404.xsl]: Please revert

File: workspace/utilities/com/erreurs.xsl [hunk 20]
```diff
@@ -1,14 +0,0 @@
-<?xml version="1.0" encoding="UTF-8"?>
```
Comment (nitriques) [inline workspace/utilities/com/erreurs.xsl]: Please keep this

File: workspace/utilities/com/modules/module.xsl [hunk 21]
```diff
@@ -0,0 +1,50 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (nitriques) [inline workspace/utilities/com/modules/module.xsl]: What is this ?

File: workspace/utilities/lib/svg-icons.xsl [hunk 22]
```diff
@@ -1,45 +0,0 @@
-<?xml version="1.0" encoding="UTF-8"?>
```
Comment (nitriques) [inline workspace/utilities/lib/svg-icons.xsl]: Why delete this ?

File: workspace/assets/js.json (hunk 4 above)
Comment (dominic-blain) [reply workspace/assets/js.json]: You are right video.js should not be here but youtube and vimeo should be there by default in the starter kit in 2019 (2020?). They work with module-oembed and every site should include one.

File: workspace/assets/css/vars/theme.xml (hunk 7 above)
Comment (dominic-blain) [reply workspace/assets/css/vars/theme.xml]: Yes, all unused are now commented.

File: workspace/jit-image-manipulation/trusted-sites (hunk 17 above)
Comment (dominic-blain) [reply workspace/jit-image-manipulation/trusted-sites]: This is a result of all the shuffling around this PR as been through, not my intention.

File: workspace/utilities/com/modules/module.xsl (hunk 21 above)
Comment (dominic-blain) [reply workspace/utilities/com/modules/module.xsl]: Starting point for all modules components. Keeps everything normalized and easy to maintain.

File: workspace/utilities/lib/svg-icons.xsl (hunk 22 above)
Comment (dominic-blain) [reply workspace/utilities/lib/svg-icons.xsl]: Has not been used since 1983. But mostly because we also had site-svg and one of them was unnecessary. Chose this one.

File: install/includes/config_default.php (hunk 6 above)
Comment (dominic-blain) [reply install/includes/config_default.php]: I prefer being explained and thanking you now ;)

File: workspace/assets/js.json (hunk 4 above)
Comment (nitriques) [reply workspace/assets/js.json]: @dominic-blain The thing is this: it is easier to see that the js file is missing, than see that it is unused...

So I would rather have them in the doc section of each xlt file.

File: install/includes/config_default.php (hunk 6 above)
Comment (nitriques) [reply install/includes/config_default.php]: It's a surprise. It will do the same thing (feature wise)

File: workspace/utilities/com/modules/module.xsl (hunk 21 above)
Comment (nitriques) [reply workspace/utilities/com/modules/module.xsl]: Is it imported by default ?

Maybe we should put "templates" in another folder ?

File: workspace/data-sources/data.navigation.php [hunk 23]
```diff
@@ -0,0 +1,42 @@
+<?php
+
+// require_once(EXTENSIONS.'/page_lhandles/lib/class.datasource.MultilingualNavigation.php');
+
+class datasourcenavigation extends MultilingualNavigationDatasource
```
Comment (nitriques) [inline workspace/data-sources/data.navigation.php]:
```suggestion
if (!class_exists('MultilingualNavigationDatasource')) {
    class MultilingualNavigationDatasource extends NavigationDatasource {}
}

class datasourcenavigation extends MultilingualNavigationDatasource
```

File: workspace/utilities/com/modules/module.xsl (hunk 21 above)
Comment (dominic-blain) [reply workspace/utilities/com/modules/module.xsl]: Every modules calls this template so yes, it needs to be imported by default for all modules to work. I'm not against a new folder, but I also think it being in the "Modules" folder works too. Maybe the kind of thing we can think about after this PR is succesfull, hehe!

File: workspace/assets/js.json [hunk 24]
```diff
@@ -61,11 +60,7 @@
 		"https://ajax.googleapis.com/ajax/libs/jquery/2.2.4/jquery.min.js",
 		"https://cdnjs.cloudflare.com/ajax/libs/jquery-easing/1.3/jquery.easing.min.js",
 		"https://cdnjs.cloudflare.com/ajax/libs/underscore.js/1.8.3/underscore-min.js",
-		"https://cdnjs.cloudflare.com/ajax/libs/underscore.string/3.3.4/underscore.string.min.js",
-		"https://cdnjs.cloudflare.com/ajax/libs/velocity/1.5.0/velocity.min.js",
```
[the remark this answers is not in the document]
Comment (dominic-blain) [reply workspace/assets/js.json]: Added back!

File: install/includes/config_default.php (hunk 6 above)
Comment (dominic-blain) [reply install/includes/config_default.php]: You just want it to be gitignored ?

File: workspace/assets/js.json (hunk 4 above)
Comment (dominic-blain) [reply workspace/assets/js.json]: I understand that, and that's why I removed video.js since it could be unused but my point was that oembed must always be in every site. So basic things like this are the minimum.

But if you still want to remove it that will be my last plea against it.

File: workspace/utilities/com/modules/module.xsl (hunk 21 above)
Comment (nitriques) [reply workspace/utilities/com/modules/module.xsl]: I'll keep it for now, but this is a performance hit. Do not be afraid to copy paste things. It it sometimes more performant and more flexible than having layers and layers of template that do not have a clear responsibility.

File: workspace/assets/js.json (hunk 4 above)
Comment (dominic-blain) [reply workspace/assets/js.json]: Done!

File: install/includes/config_default.php (hunk 6 above)
Comment (nitriques) [reply install/includes/config_default.php]: @dominic-blain no, justed renamed/moved under  `manifest/unattended.php`

File: workspace/assets/js.json [hunk 25]
```diff
@@ -28,12 +27,14 @@
 		"modules/page-load-errors.js",
 		"modules/page-load.js",
 		"modules/change-state.js",
-		"modules/auto-jit-image.js",
```
Comment (nitriques) [inline workspace/assets/js.json]: If we remove this, we can also remove both files in lib ("lib/jquery.sizing.min.js", "lib/jquery.jit-image.min.js")

File: install/includes/config_default.php (hunk 6 above)
Comment (dominic-blain) [reply install/includes/config_default.php]: Great!

File: workspace/assets/js.json (hunk 25 above)
Comment (dominic-blain) [reply workspace/assets/js.json]: Done!
